src/test/xmeta/CMakeLists.txt

```diff
@@ -27,7 +42,13 @@ if (MSVC)
     target_link_libraries(test_xmeta windowsapp ole32)
 endif()
 
-target_sources(test_xmeta PUBLIC main.cpp)
+# TODO Refactor the tool dependencies into a library that we can simply link against (instead of double compiling)
```

Comment:
Remove these dependencies on cpp files in the tool. Instead, put that functionality into a lib that you can link the test against.